x/mongo/driver/topology/server.go

```diff
@@ -469,16 +470,6 @@ func (s *Server) updateDescription(desc description.Server, initial bool) {
 		c <- desc
 	}
 	s.subLock.Unlock()
-
-	if initial {
-		// We don't clear the pool on the first update on the description.
-		return
-	}
-
-	switch desc.Kind {
```

Comment:
Is there no other time, other than when we manually set the kind to unknown on line 258, that the kind becomes unknown and the pool should be cleared?

Reply:
a lot of the places that should clear when the kind is set to unknown were actually double clearing before by calling both updateDescription (which drained) and pool.clear, so now those places are only clearing once